google/cloud/aiplatform/prediction/local_endpoint.py

```diff
@@ -0,0 +1,329 @@
+
+    def _stop_container_if_exists(self):
+        """Stops the container if the container exists."""
+        if self.container is not None:
+            self.container.stop()
+
+    def predict(
+        self,
+        request: Optional[Any] = None,
+        request_file: Optional[str] = None,
+        headers: Optional[Dict] = None,
+    ) -> requests.models.Response:
+        """Executes a prediction.
+
+        Args:
+            request (Any):
+                Optional. The request sent to the container.
+            request_file (str):
+                Optional. The path to a request file sent to the container.
+            headers (Dict):
+                Optional. The headers in the prediction request.
+
+        Returns:
+            The response from the prediction; None if the prediction raises exception.
+
+        Raises:
+            ValueError: If both of request and request_file are specified, both of
+                request and request_file are not provided, or request_file is specified
+                but does not exist.
+            requests.exception.RequestException: If the request fails with an exception.
+        """
+        if request is not None and request_file is not None:
+            raise ValueError(
+                "request and request_file can not be specified at the same time."
+            )
+        if request is None and request_file is None:
+            raise ValueError("One of request and request_file needs to be specified.")
+
+        try:
+            url = f"http://localhost:{self.host_port}{self.serving_container_predict_route}"
+            if request is not None:
+                response = requests.post(url, data=request, headers=headers)
+            elif request_file is not None:
+                if not Path(request_file).expanduser().resolve().exists():
+                    raise ValueError(f"request_file does not exist: {request_file}.")
+                with open(request_file) as data:
+                    response = requests.post(url, data=data, headers=headers)
+            return response
+        except requests.exceptions.RequestException as exception:
+            _logger.warning(f"Exception during health check: {exception}")
+            raise
+
+    def run_health_check(self) -> requests.models.Response:
+        """Runs a health check.
+
+        Returns:
+            The response from the health check or None if the health check raises exception.
+
+        Raises:
+            requests.exception.RequestException: If the request fails with an exception.
+        """
+        try:
+            url = f"http://localhost:{self.host_port}{self.serving_container_health_route}"
+            response = requests.get(url)
+            return response
+        except requests.exceptions.RequestException as exception:
+            _logger.warning(f"Exception during health check: {exception}")
+            raise
+
+    def print_container_logs(self, show_all: bool = False):
```

Comment:
Are these print log methods meant to be called directly by the user?

Reply:
It can be explicitly called by users or called if anything is wrong during local docker runs.